hw2/arturpendrag0n.cs

```
@@ -0,0 +1,202 @@
+            }
+            return array;
+        }
+        static int[] SortDescending(int[] sourceArray)
+        {
+            int[] array = returnArrayCopy(sourceArray);
+            int temp;
+            for (int i = 0; i < array.Length; i++)
+            {
+                for (int j = 0; j < array.Length - 1; j++)
+                {
+                    if (array[j] < array[j + 1])
+                    {
+                        //we store the bigger number in temp
+                        temp = array[j + 1];
+                        //we swap the places of the two numbers
+                        array[j + 1] = array[j];
+                        array[j] = temp;
+                    }
+                }
+            }
+            return array;
+        }
+        static int[] returnArrayCopy(int[] sourceArray) 
+        {
+            int[] array = new int[sourceArray.Length];
+            Array.Copy(sourceArray, array, sourceArray.Length);
+            return array;
+        }
+        static int[] addAtEnd(int[] array, int element)
+        {
+            int[] expandedArray = new int[array.Length+1];
+            for(int i = 0; i < array.Length; i++)
+            {
+                expandedArray[i] = array[i];
+            }
+            expandedArray[array.Length] = element;
+            return expandedArray;
+        }
+        static int[] addAtBeginning(int[] array,int element)
+        {
+            int[] expandedArray = new int[array.Length + 1];
+            expandedArray[0] = element;
+            for (int i = 0; i < array.Length; i++)
+            {
+                expandedArray[i+1] = array[i];
+            }
+            return expandedArray;
+        }
+        static int[] addAtAny(int[] array, int element,int position)
+        {
+            if (position < 0 || position > array.Length)
+            {
+                Console.WriteLine($"This position is out of the array's bounds please choose between {0} and {array.Length}");
```

Good idea to check here.  In the future try throwing an exception instead of printing a message.  I think Kai is covering those in today's lesson.